controller/build/build.go

```diff
@@ -42,41 +43,10 @@ import (
 const defaultTargetName = "default"
 
 func RunBuild(ctx context.Context, dockerCli command.Cli, in controllerapi.BuildOptions, inStream io.Reader, progressMode string, statusChan chan *client.SolveStatus) (*client.SolveResponse, *build.ResultContext, error) {
-	if in.Opts.NoCache && len(in.NoCacheFilter) > 0 {
-		return nil, nil, errors.Errorf("--no-cache and --no-cache-filter cannot currently be used together")
-	}
-
-	contexts := map[string]build.NamedContext{}
-	for name, path := range in.NamedContexts {
-		contexts[name] = build.NamedContext{Path: path}
-	}
-
-	printFunc, err := parsePrintFunc(in.PrintFunc)
+	opts, err := ToBuildOpts(in, inStream)
```

Comment:
It seems that some fields (e.g. Exports) are processed twice (in ToBuild and in RunBuild), can we try to dedupe them?